morango/utils/sync_utils.py

```diff
@@ -25,6 +25,21 @@ def _self_referential_fk(klass_model):
                 return f.attname
     return None
 
+def _fsic_queuing_calc(fsic1, fsic2):
```

Comment:
I've been playing with a one-line version of this that may be a bit more readable (that may be subjective, but the deletion of keys threw me at first in the current version):

```
return {instance: fsic2.get(instance, 0) for instance, counter in iteritems(fsic1) if fsic2.get(instance, 0) < counter}
```

I did some speed benchmarks, and it seems to depend. With a small number of keys (< 1000), your version is substantially faster. With a large number of keys, my one-liner is faster. When the two dictionaries are identical (or don't have many differences), my version seems to be faster (probably because you're starting with a full dict and then deleting everything in it one by one). The numbers we're talking about are small (it's a fast function in any case), and since it may be fairly common for the fsics to be largely the same (with legacy instance IDs sticking around but not changing), you might consider the one-liner if it doesn't feel too obtuse. Alternatively, here's a few-liner that performs similarly:

```
    fsic = {}
    for instance, counter in iteritems(fsic1):
        fsic2counter = fsic2.get(instance, 0)
        if fsic2counter < counter:
            fsic[instance] = fsic2counter
    return fsic
```

Reply:
Probably good to throw a couple simple unit tests at this function, too.

Reply:
I have integration tests that actually test this functionality.